elfloader-tool/src/arch-riscv/boot.c

```diff
@@ -28,14 +35,28 @@
 // page table entry (PTE) field
 #define PTE_V     0x001 // Valid
 
-#define PTE64_PPN0_SHIFT 10
+#define PTE_PPN0_SHIFT 10
 
-#define PTES_PER_PT (RISCV_PGSIZE/sizeof(long))
+#if __riscv_xlen == 32
+#define PTE_LEVEL_BITS PT_LEVEL_2_BITS
+#define PT_INDEX_BITS  10
+#define PPN_HIGH       20
+typedef uint32_t seL4_Word;
+#else
+#define PTE_LEVEL_BITS PT_LEVEL_1_BITS
+#define PT_INDEX_BITS  9
+#define PPN_HIGH       28
+typedef uint64_t seL4_Word;
+#endif
+
+#define PTES_PER_PT (1 << PT_INDEX_BITS)
```

Comment:
nitpick: please use the BIT macro here, as it

Reply:
I'll update to use the BIT macro.